client/profile.js

```diff
@@ -8,10 +8,9 @@ export default class Profile extends Component {
     this.state = {
       puppy: {}
     }
-    this.profile = this.profile.bind(this)
   }
   async componentDidMount () {
-    const puppyId = this.props.match.params.puppyId
+    const puppyId = this.props.match.params.profileId
```

Comment:
You set up your router to pass down a `profileId`, not `puppyId` (see the routes section of your frontend). This was all that prevented this profile component from working properly.